### PR TITLE
Display premium features tabs on the dashboard

### PR DESCRIPTION
**Pull request by vinibrsl**
This commit makes the funnels and props tab on the dashboard visible even when the site owner's plan does not have access to that feature. This is to raise awareness of those features, and if the site owner does not want to see that tab, they can still click "Hide this report" to hide it.

Previously, there was no way of differentiating between opting out of a feature or not having access to it, because `Plausible.Billing.Feature.enabled/1` returns `false`, regardless of the `sites.feature_enabled` toggle. This commit creates a new function called `opted_out/1` to differentiate access from explicitly opting out a feature.

### CODE REVIEW

File: lib/plausible/billing/feature.ex [hunk 1]
```diff
@@ -109,6 +116,11 @@ defmodule Plausible.Billing.Feature do
         end
       end
 
+      @impl true
+      def opted_out?(%Plausible.Site{} = site) do
+        if is_nil(toggle_field()), do: false, else: not Map.fetch!(site, toggle_field())
+      end
```
Comment (RobertJoonas) [inline lib/plausible/billing/feature.ex:122]: The same condition is actually repeated on lines 114 and 115. What do you think about rewriting the cond clause into this instead:

```elixir
check_availability(site.owner) == :ok && not opted_out?(site)
```

**Review by RobertJoonas — CHANGES_REQUESTED** [review]
I tested it out on staging and it looks like we're currently not allowing the user to opt out of a feature that they don't have access to.

I clicked on the "Hide this report" button (for Funnels) in the feature setup notice and nothing happened (I was on a Growth plan). I expected the feature notice setup to disappear.

Left some comments inline as well with some suggestions.

File: lib/plausible/billing/feature.ex (hunk 1 above)
Comment (vinibrsl) [reply lib/plausible/billing/feature.ex]: Good call! Changed in 81fdd861e574026e02241b414d81af7d1f8b06b1

File: lib/plausible_web/templates/stats/stats.html.eex [hunk 2]
```diff
@@ -18,9 +18,12 @@
     data-domain="<%= @site.domain %>"
     data-offset="<%= Plausible.Site.tz_offset(@site) %>"
     data-has-goals="<%= @has_goals %>"
-    data-conversions-enabled="<%= Plausible.Billing.Feature.Goals.enabled?(@site) %>"
-    data-funnels-enabled="<%= Plausible.Billing.Feature.Funnels.enabled?(@site) %>"
-    data-props-enabled="<%= Plausible.Billing.Feature.Props.enabled?(@site) %>"
+    data-conversions-opted-out="<%= Plausible.Billing.Feature.Goals.opted_out?(@site) %>"
+    data-funnels-opted-out="<%= Plausible.Billing.Feature.Funnels.opted_out?(@site) %>"
+    data-props-opted-out="<%= Plausible.Billing.Feature.Props.opted_out?(@site) %>"
+    data-conversions-available="<%= Plausible.Billing.Feature.Goals.check_availability(@site.owner) == :ok %>"
```
[the remark this answers is not in the document]
Comment (vinibrsl) [reply lib/plausible_web/templates/stats/stats.html.eex]: Changed in d1025b1e7bb7702bf46e37db2875bfd381ed7055

File: assets/js/dashboard.js [hunk 3]
```diff
@@ -16,10 +16,13 @@ if (container) {
     domain: container.dataset.domain,
     offset: container.dataset.offset,
     hasGoals: container.dataset.hasGoals === 'true',
-    conversionsEnabled: container.dataset.conversionsEnabled === 'true',
-    funnelsEnabled: container.dataset.funnelsEnabled === 'true',
-    propsEnabled: container.dataset.propsEnabled === 'true',
     hasProps: container.dataset.hasProps === 'true',
+    conversionsAvailable: container.dataset.conversionsAvailable === 'true',
```
[the remark this answers is not in the document]
Comment (vinibrsl) [reply assets/js/dashboard.js]: Changed in d1025b1e7bb7702bf46e37db2875bfd381ed7055

**Comment by vinibrsl** [comment]
> I clicked on the "Hide this report" button (for Funnels) in the feature setup notice and nothing happened (I was on a Growth plan). I expected the feature notice setup to disappear.

Oops! I fixed this bug in dc97e83de412a08a5f77cf77f3ac2d56b615d060.

**Review by RobertJoonas — APPROVED** [review]

**Pull request merged**